tests/Server/TextDocument/SignatureHelpTest.php

```diff
@@ -0,0 +1,222 @@
+    }
+
+    public function testMethodClosedReference()
+    {
+        $completionUri = pathToUri(__DIR__ . '/../../../fixtures/signatureHelp/methodClosed.php');
+        $this->loader->open($completionUri, file_get_contents($completionUri));
+        $result = $this->textDocument->signatureHelp(
+            new TextDocumentIdentifier($completionUri),
+            new Position(14, 11)
+        )->wait();
+
+        $help = new SignatureHelp;
+        $help->signatures = [];
+        $info = new SignatureInformation;
+        $help->signatures[] = $info;
+        $info->label = 'method(string $param = "")';
+        $info->parameters = [];
+        $param = new ParameterInformation;
+        $info->parameters[] = $param;
+        $param->label = 'string $param = ""';
+
+        $this->assertEquals($help, $result);
+    }
+
+    public function testMethodNotClosed()
+    {
+        $completionUri = pathToUri(__DIR__ . '/../../../fixtures/signatureHelp/methodNotClosed.php');
+        $this->loader->open($completionUri, file_get_contents($completionUri));
+        $result = $this->textDocument->signatureHelp(
+            new TextDocumentIdentifier($completionUri),
+            new Position(9, 22)
+        )->wait();
+
+        $help = new SignatureHelp;
+        $help->signatures = [];
+        $info = new SignatureInformation;
+        $help->signatures[] = $info;
+        $info->label = 'method(string $param = "")';
+        $info->parameters = [];
+        $param = new ParameterInformation;
+        $info->parameters[] = $param;
+        $param->label = 'string $param = ""';
+
+        $this->assertEquals($help, $result);
+    }
+
+    public function testMethodNotClosedReference()
+    {
+        $completionUri = pathToUri(__DIR__ . '/../../../fixtures/signatureHelp/methodNotClosed.php');
+        $this->loader->open($completionUri, file_get_contents($completionUri));
+        $result = $this->textDocument->signatureHelp(
+            new TextDocumentIdentifier($completionUri),
+            new Position(14, 14)
+        )->wait();
+
+        $help = new SignatureHelp;
+        $help->signatures = [];
+        $info = new SignatureInformation;
+        $help->signatures[] = $info;
+        $info->label = 'method(string $param = "")';
+        $info->parameters = [];
+        $param = new ParameterInformation;
+        $info->parameters[] = $param;
+        $param->label = 'string $param = ""';
+
+        $this->assertEquals($help, $result);
+    }
+
+    public function testFuncClosed()
+    {
+        $completionUri = pathToUri(__DIR__ . '/../../../fixtures/signatureHelp/funcClosed.php');
+        $this->loader->open($completionUri, file_get_contents($completionUri));
+        $result = $this->textDocument->signatureHelp(
+            new TextDocumentIdentifier($completionUri),
+            new Position(6, 10)
+        )->wait();
+
+        $help = new SignatureHelp;
+        $help->signatures = [];
+        $info = new SignatureInformation;
```

Comment:
Add constructors to the classes that allow to pass all properties as arguments so we can define the result inline without saving it to a variable